### DISCUSSION TITLE
Using PSReadLine on hardened workstations

### DISCUSSION THREAD

**Discussion by erschef** [lzybkr/PSReadLine#3236] [Q&A] [answered]
Hello all,

I'm inquiring to my security team about enabling the use of PSReadLine on our hardened management workstations, and am being asked:
-Does the module require admin rights to run?
-Can the module be run in Constrained Language Mode?
Interested to hear what folks think, and or if there is documentation stating this that I'm not aware of. Thanks!

**Comment by daxian-dbw** [comment, marked as answer]
The module doesn't need admin rights to run.
The module has to run in full language mode, so if your environment is constrained with AppLocker or so, you will have to put this module in the allow-list somehow. /cc @TravisEz13 for additional insights.

**Reply by erschef** [reply]
Thanks for the feedback; I did some quick, limited testing on my own and found that the module continued to work after setting CLM on my machine, but I can't recall if I tried closing and re-opening the window to see if it continued to work in a new session. I'll pass this information back to my security team, thanks for the information.
